pkg/client/entities_client.go

```diff
@@ -14,5 +14,191 @@
 
 package client
 
+import (
+	"encoding/json"
+	"net/url"
+	reflect "reflect"
+	"regexp"
+	"unicode"
+
+	"github.com/dynatrace/dynatrace-configuration-as-code/internal/throttle"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/rest"
+)
+
 const pathEntitiesObjects = "/api/v2/entities"
 const pathEntitiesTypes = "/api/v2/entityTypes"
+
+// Full export would be: const defaultListEntitiesFields = "+lastSeenTms,+firstSeenTms,+tags,+managementZones,+toRelationships,+fromRelationships,+icon,+properties"
+// Using smaller export for faster processing, using less memory
+const defaultListEntitiesFields = "+lastSeenTms,+firstSeenTms"
+
+var extraEntitiesFields = map[string][]string{
+	"properties":      {"detectedName", "oneAgentCustomHostName", "ipAddress"},
+	"toRelationships": {"isSiteOf", "isClusterOfHost"},
+}
+
+func getEntitiesTypeFields(entitiesType EntitiesType, ignoreProperties []string) string {
+	typeFields := defaultListEntitiesFields
+
+	for topField, subFieldList := range extraEntitiesFields {
+
+		if contains(ignoreProperties, topField) {
+			continue
+		}
+
+		fieldSliceObject := getDynamicFieldFromObject(entitiesType, topField)
+
+		if isInvalidReflectionValue(fieldSliceObject) {
+			continue
+		} else {
```

Comment:
You can actually drop the else block completely now. 
If the value is invalid, we already continue the loop/exit early.
So the rest of method can just be the normal/happy path case.

```go
		if isInvalidReflectionValue(fieldSliceObject) {
			continue
		}
		
		for _, subField := range subFieldList {
			if contains(ignoreProperties, subField) {
				continue
			}
			if hasSpecificFieldValueInSlice(fieldSliceObject, "id", subField) {
				typeFields = typeFields + ",+" + topField + "." + subField
			}
		}
```